### ISSUE TITLE
Implement `IntoIterator` for fixed size arrays

### ISSUE THREAD

**Issue by nrc** [rust-lang/rust#29455]
Size 0 to 12 or whatever.

Because otherwise it is only implemented via autoref and then you get references to the contents where you want the contents by value.

**Comment by Gankro** [comment]
psuedo-blocked on ManuallyDrop https://github.com/rust-lang/rfcs/pull/197

There's no way to soundly partially move out of an array in a zero-cost way today. @bluss' [ArrayVec](https://github.com/bluss/arrayvec) crate accomplishes this by putting the array in a modified Option (one that supresses the null ptr optimization) and ptr::writeing None in the dtor.

**Comment by Gankro** [comment]
We can certainly do the ArrayVec thing to get the job done if it's dire.

**Comment by bluss** [comment]
This would be a breaking change since `[1, 2, 3].into_iter()` compiles in Rust 1.4. However it's a minor change by being avoidable with more specific/UFCS method calls.

Fortunately `for _ in [1, 2, 3]` does not compile in Rust 1.4.

**Comment by Gankro** [comment]
Seems like acceptable breakage, since this is clearly desirable.

**Comment by steveklabnik** [comment]
This is a duplicate of https://github.com/rust-lang/rust/issues/25725
